src/runtime_src/core/common/device.h

```diff
@@ -511,6 +511,28 @@ device_query(const std::shared_ptr<device>& device, Args&&... args)
   return boost::any_cast<typename QueryRequestType::result_type>(ret);
 }
 
+template <typename QueryRequestType>
+inline typename QueryRequestType::result_type
+device_query_default(const device* device, const typename QueryRequestType::result_type& default_value = typename QueryRequestType::result_type())
```

Comment:
I like these functions, but I am not sure I like the default initialization of the `default_argument`.  If calling code is relying on a default value, I think it should be passed in explicitly.